docs/commands/deploy.md

```diff
@@ -88,7 +88,7 @@ netlify deploy
 - `functions` (*string*) - Specify a functions folder to deploy
 - `prod` (*boolean*) - Deploy to production
 - `prodIfUnlocked` (*boolean*) - Deploy to production if unlocked, create a draft otherwise
-- `alias` (*string*) - Specifies the alias for deployment. Useful for creating predictable deployment URLs. Maximum 37 characters.
+- `alias` (*string*) - Specifies the alias for deployment, the string at the beginning of the deploy subdomain. Useful for creating predictable deployment URLs. Avoid setting an alias string to the same value as a deployed branch. `alias` doesn’t create a branch deploy and can’t be used in conjunction with the branch subdomain feature. Maximum 37 characters.
```

Comment:
Reviewer note: this is a lot longer than our normal description, but I'm attempting to help users avoid misconceptions about usage for this flag.